### PR TITLE
✨  Add support for Chorus context

### PR DESCRIPTION
**Pull request by benoitgrelard**
This PR adds support for Chorus context via `chorus-js`.

🔥  Breaking change: `publishStep` signature has changed from `(...args: Array<string>) => number | string | void` to `(Array<string>, Context) => number | string | void`

This will be published as a major release because of the breaking change.

### CODE REVIEW

File: src/__tests__/createContext.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		});
+	});
+
+	describe('toObject', () => {
+		it('outputs as an object', () => {
+			const obj = context.toObject();
+			expect(obj).toEqual({ a: 1, b: 2 });
+		});
+	});
+});
```
Comment (peduarte) [inline src/__tests__/createContext.js:52]: Maybe worth adding a test that fails when you try to modify a property of the context via dot notation?

Comment (benoitgrelard) [reply src/__tests__/createContext.js:52]: Done

**Comment by benoitgrelard** [comment]
@nickebbutt Can you let me know if you want to review this before we merge?

**Review by nickebbutt — APPROVED** [review]
Looks good!

File: example/createCounter.js [hunk 2]
```diff
@@ -36,38 +36,42 @@ export default function (rootElem: ?HTMLElement): Counter {
...
 		});
-		client.publishStep('.* call a step which times out', () => {
+		client.publishStep('.*call a step which times out', () => {
 			setTimeout(() => { _fakeErrorValue = true; }, 1000);
 			expect(_fakeErrorValue).toBe(true);
 			return String(_fakeErrorValue);
 		}, {
 			retryDuration: 500,
 		});
+		client.publishStep('.*in chorus-js \'(.*)\' has the value \'(.*)\'', ([name, value], context) => {
```
Comment (nickebbutt) [inline example/createCounter.js:61]: Can you explain to me, is the [name, value] parameter an associative array?

Comment (benoitgrelard) [reply example/createCounter.js:61]: The function simply receives as the first argument the arguments sent from chorus as an array. The syntax you see here is part of ES6 and is called `destructuring`. I'm `"picking"` out the 2 first values out of the array and naming them right there in the function parameters.

I could have done this instead:

```
client.publishStep('.*in chorus-js \'(.*)\' has the value \'(.*)\'', (args, context) => {
    const [name, value] = args;
    console.log(name, value); // I now have access to these directly
    ....
```

Comment (nickebbutt) [reply example/createCounter.js:61]: Thanks for explaining that!

**Pull request merged**